src/Controller/Api/ApiController.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Buddy\Repman\Controller\Api;
+
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\Form\FormError;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+
+abstract class ApiController extends AbstractController
+{
+    protected function createApiForm(string $class): FormInterface
+    {
+        return $this->createForm(
+            $class, null, [
+                'allow_extra_fields' => true,
+                'csrf_protection' => false,
+            ]
+        );
+    }
+
+    /**
+     * @return array<string,mixed>
+     */
+    protected function parseJson(Request $request): array
+    {
+        if ($request->getContent() === '') {
+            return [];
+        }
+
+        $data = json_decode($request->getContent(), true);
```

Comment:
Why not take an advantage of JSON_THROW_ON_ERROR flag available since PHP 7.3? 🙃